mesh/pumi.cpp

```diff
@@ -1348,279 +1138,148 @@ void ParPumiMesh::FieldMFEMtoPUMI(apf::Mesh2* apf_mesh,
-         // Set vel
-         double vels[3] = {u_vel[kk], v_vel[kk], w_vel[kk]};
-         apf::setComponents(VelField, vtxs[kk], 0, vels);
-      }
-
-      int dofId = num_vts;
+      ElementTransformation* eltr = this->GetElementTransformation(iel);
+      DenseMatrix vel;
+      grid_vel->GetVectorValues(*eltr, mfem_nodes, vel);
 
-      apf::EntityShape* es = VelFieldShape->getEntityShape(apf::Mesh::TET);
-      // Edge Dofs
-      if (VelFieldShape->hasNodesIn(apf::Mesh::EDGE))
+      int non = 0;
+      for (int d = 0; d <= dim; d++)
       {
-         int ndOnEdge = VelFieldShape->countNodesOn(apf::Mesh::EDGE);
-         Array<int> order(ndOnEdge);
-
-         apf::Downward edges;
-         int num_edge =  apf_mesh->getDownward(ent, apf::Mesh::EDGE, edges);
-         for (int ii = 0 ; ii < num_edge; ++ii)
+         if (!field_shape->hasNodesIn(d)) { continue; }
+         apf::Downward a;
+         int na = apf_mesh->getDownward(ent,d,a);
+         for (int i = 0; i < na; i++)
          {
-            es->alignSharedNodes(apf_mesh, ent, edges[ii], order);
-            for (int jj = 0; jj < ndOnEdge; jj++)
+            int type = apf_mesh->getType(a[i]);
+            int nan = field_shape->countNodesOn(type);
+            for (int n = 0; n < nan; n++)
             {
-               int cnt = dofId + order[jj];
-               double mag = u_vel[cnt] * u_vel[cnt] +
-                            v_vel[cnt] * v_vel[cnt] +
-                            w_vel[cnt] * w_vel[cnt];
-               mag = sqrt(mag);
-               apf::setScalar(VelMagField, edges[ii], jj, mag);
-
-               // Set vel
-               double vels[3] = {u_vel[cnt], v_vel[cnt], w_vel[cnt]};
-               apf::setComponents(VelField, edges[ii], jj, vels);
+               apf::Vector3 v(vel.GetColumn(non));
+               apf::setScalar(vel_mag_field, a[i], n, v.getLength());
+               apf::setVector(vel_field, a[i], n, v);
+               non++;
             }
-            // Counter
-            dofId += ndOnEdge;
          }
       }
+      iel++;
+   }
+   apf_mesh->end(itr);
+}
 
-      // Face Dofs
-      if (VelFieldShape->hasNodesIn(apf::Mesh::TRIANGLE))
-      {
-         int ndOnFace = VelFieldShape->countNodesOn(apf::Mesh::TRIANGLE);
-         Array<int> order(ndOnFace);
+void ParPumiMesh::NedelecFieldMFEMtoPUMI(apf::Mesh2* apf_mesh,
+                                         ParGridFunction* gf,
+                                         apf::Field* nedelec_field)
+{
+   apf::FieldShape* nedelecFieldShape = nedelec_field->getShape();
+   int dim = apf_mesh->getDimension();
 
-         apf::Downward faces;
-         int num_face = apf_mesh->getDownward(ent, apf::Mesh::TRIANGLE, faces);
-         for (int ii = 0; ii < num_face; ii++)
+   // loop over all elements
+   size_t elemNo = 0;
+   apf::MeshEntity* ent;
+   apf::MeshIterator* it = apf_mesh->begin(dim);
+   while ( ent = apf_mesh->iterate(it) )
```

Comment:
Is this intended to be an assignment? If yes can you use double parenthesis so that the compiler warning is suppressed?

Reply:
fixed in https://github.com/mortezah/mfem/commit/6f6e134d45b5a39f556cecfa284396cb515795f6